ingest-app/src/cmr/ingest/api/generic_documents.clj

```diff
@@ -129,10 +130,15 @@
          (pull-metadata-specific-information request-context concept-type content-type raw-document)]
     ;; Check to see if the passed in record contains the MetadataSpecification/Name field and its
     ;; value matches that from the concept name in the route parameters.
-    (if (and (not= concept-type spec-key)
-             (not= (concept-type draft-concept->spec-map) spec-key)
-             (not= (common-concepts/get-concept-type-of-draft concept-type) spec-key))
-      (throw (UnsupportedOperationException. (format "%s version %s is not supported." spec-key spec-version)))
+    (if (or (not spec-key)
+            (and (not= concept-type spec-key)
+                 (not= (concept-type draft-concept->spec-map) spec-key)
+                 (not= (common-concepts/get-concept-type-of-draft concept-type) spec-key)))
+      (errors/throw-service-error
+       :invalid-data
+       (if-not spec-key
+         "MetadataSpecification is missing"
```

Comment:
Can you change the string to 

"The MetadataSpecification schema element is missing from the record being ingested."

Reply:
Done